rust/private/rustdoc_test.bzl

```diff
@@ -56,6 +56,11 @@ def _construct_writer_arguments(ctx, test_runner, action, crate_info, rust_toolc
     for var in action.env.keys():
         writer_args.add("--action_env={}".format(var))
 
+    # Ensure the rustdoc binary is always accessed via a relative path
```

Comment:
Good call. The path to rustdoc moved from being just a path to some file in `external` to a `bazel-out/*` directory since the file was now generated from some action. I've corrected the `rustdoc_compile_action` macro to take a flag `is_test` so the action can correctly be configured for `rust_doc_test` targets.